src/LowMachEquationSystem.C

```diff
@@ -1739,55 +1738,70 @@ MomentumEquationSystem::register_wall_bc(
   VectorFieldType *theBcField = &(meta_data.declare_field<VectorFieldType>(stk::topology::NODE_RANK, bcFieldName));
   stk::mesh::put_field_on_mesh(*theBcField, *part, nDim, nullptr);
 
-  // extract the value for user specified velocity and save off the AuxFunction
-  AuxFunction *theAuxFunc = NULL;
-  std::string velocityName = "velocity";
+  if(realm_.solutionOptions_->meshMotion_) {
```

Comment:
@sayerhs So currently in the code if there is a wall boundary condition in the input file,`wall_user_data` is a must. The definition of `wall_user_data` in the input file can be empty but there always needs to be a `wall_user_data` key word. So for a user to move from an auxiliary function based definition of `velocity` at the wall to using the `mesh_velocity` they would need to remember to remove the key word "velocity" from the `wall_user_data` in the input file but not `wall_user_data` itself. This seems very prone to user-induced errors. What do you think ? I guess it comes down to the question you asked, why and when would one not want a no-slip condition along the moving interfaces ?

Reply:
So two issues I notice: 1. how will this formulation affect when the user wants to use wall function, and 2. how will this affect where there are moving and non-moving walls? Terrain with rotating turbines comes to mind. Are we taking care of registering `mesh_velocity` field on all parts even if they are not registered through mesh_motion section so that the non-moving walls essentially have a no-op?

Reply:
@sayerhs 

1) This formulation currently prevents the user from using a wall function for the momentum equation if mesh motion has been specified. I see that being undesirable.  How about we let the user specify `velocity: mesh_motion` under `wall_user_data` ? Choosing this option would just copy the mesh velocity over to the fluid velocity. While this does introduce an additional variable for the user to set, it would allow the user to choose how the velocity at the wall is handled. If you agree with this approach, please let me know and I can make the appropriate changes. 

2) The mesh motion algorithm does not handle registration of `mesh_velocity`. In the scenario that there is mesh motion, Realm::register_nodal_fields() registers `mesh_velocity` for all part names returned by Realm::get_physics_target_names().

Reply:
OK. Let's leave the code as you have it right now and revisit if we see a use-case where we would want a different behavior.